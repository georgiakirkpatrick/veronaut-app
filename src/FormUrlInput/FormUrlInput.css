.FormUrlInput {
    width: 100%;
    margin-bottom: 2vh;
}

.FormUrlInput input {
    width: 100%;
    margin: 5px 0px 10px;
    padding: 10px;
    font: inherit;
}