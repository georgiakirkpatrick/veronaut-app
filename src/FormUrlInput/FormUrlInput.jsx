import React from 'react'
import './FormUrlInput.css'

const FormUrlInput = props => {
  const {
    currentValue,
    handleChange,
    id,
    name,
    prompt
  } = props

  return (
    <div className='FormUrlInput'>
      <label htmlFor={id}>{prompt}</label>

      <input 
        required
        type='url'
        id={id} 
        name={name}
        value={currentValue} 
        onChange={handleChange}  
      />
    </div>
  )
}

FormUrlInput.defaultProps = {
  currentValue: '',
  handleChange: () => {},
  id: 1,
  name: '',
  prompt: '',
}

export default FormUrlInput

