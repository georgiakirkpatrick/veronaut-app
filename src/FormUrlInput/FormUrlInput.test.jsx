import React from 'react'
import { render, screen } from '@testing-library/react'
import { describe, it, expect } from 'vitest'
import FormUrlInput from './FormUrlInput'

describe('FormUrlInput', () => {
    it ('renders without crashing', () => {
        render(<FormUrlInput />)
        screen.debug()
    })

    it ('render the UI as expected', () => {
        const tree = render(<FormUrlInput />)
        expect(tree).toMatchSnapshot()
    })
})