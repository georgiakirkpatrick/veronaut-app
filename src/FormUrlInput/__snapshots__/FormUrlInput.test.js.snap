// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormUrlInput render the UI as expected 1`] = `
<div
  className="FormUrlInput"
>
  <label
    htmlFor={1}
  >
    
  </label>
  <input
    id={1}
    name=""
    onChange={[Function]}
    required={true}
    type="url"
    value=""
  />
</div>
`;
